Aspire Budgeting/Facilities/GoogleSheetsManager.swift

```diff
@@ -133,4 +186,86 @@ final class GoogleSheetsManager: ObservableObject {
       }
     }
   }
+  
+  private func createSheetsValueRangeFrom(amount: String,
+                                          date: Date,
+                                          category: Int,
+                                          account: Int,
+                                          transactionType: Int,
+                                          approvalType: Int) -> GTLRSheets_ValueRange {
+    
+    let dateFormatter = DateFormatter()
+    dateFormatter.dateStyle = .medium
+    dateFormatter.timeStyle = .none
+    
+    let sheetsValueRange = GTLRSheets_ValueRange()
+    sheetsValueRange.majorDimension = kGTLRSheets_ValueRange_MajorDimension_Rows
+    sheetsValueRange.range = "Transactions!B:H"
+    
+    var valuesToInsert = [String]()
+    valuesToInsert.append(dateFormatter.string(from: date))
+    
+    if transactionType == 0 {
+      valuesToInsert.append("")
+      valuesToInsert.append(amount)
+    } else {
+      valuesToInsert.append(amount)
+      valuesToInsert.append("")
+    }
+    
+    valuesToInsert.append(transactionCategories![category])
+    valuesToInsert.append(transactionAccounts![account])
+    valuesToInsert.append("Added from Aspire iOS app")
+    
+    guard let version = self.aspireVersion else {
+      fatalError("Aspire Version is nil")
+    }
+    
+    switch version {
+    case .twoEight:
+      if approvalType == 0 {
+        valuesToInsert.append("🆗")
+      } else {
+        valuesToInsert.append("⏺")
+      }
+      
+    case .three:
+      if approvalType == 0 {
+        valuesToInsert.append("✅")
+      } else {
+        valuesToInsert.append("🅿️")
+      }
+    }
+    
+    sheetsValueRange.values = [valuesToInsert]
+    return sheetsValueRange
+  }
+  
+  func addTransaction(amount: String, date: Date, category: Int, account: Int, transactionType: Int, approvalType: Int, completion: @escaping (Bool) -> Void) {
+    
+    
+    let valuesToInsert = createSheetsValueRangeFrom(amount: amount, date: date, category: category, account: account, transactionType: transactionType, approvalType: approvalType)
+    
+    guard let authorizer = self.authorizer else {
+      self.error = GoogleDriveManagerError.nilAuthorizer
+      return
+    }
+    
+    sheetsService.authorizer = authorizer
+  
+    let appendQuery = GTLRSheetsQuery_SpreadsheetsValuesAppend.query(withObject: valuesToInsert, spreadsheetId: defaultFile!.id, range: valuesToInsert.range!)
+    
+    appendQuery.valueInputOption = kGTLRSheetsValueInputOptionUserEntered
+    
+    ticket = sheetsService.executeQuery(appendQuery, completionHandler: { (_, data, error) in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/a77c2477cb8c75fe2389ac27f223c44a1a0476c6/Aspire Budgeting/Facilities/GoogleSheetsManager.swift#L260"></span>Unused parameter “data” in a closure should be replaced with _.<br />
<code>unused_closure_parameter</code> <code>GoogleSheetsManager.swift:260</code></td>
    </tr>
  </tbody>
</table>